### PR TITLE
Make eatery id optional in reports

### PR DESCRIPTION
**Pull request by Archit404Error**
## Overview
Makes eatery_id optional in the /report endpoint because we want users to be able to report generic bugs with the app in addition to bugs with eateries, as requested by Android.

## Changes Made
Edited /report endpoint's required fields and the ReportStore model. To make eatery_id optional. Also edited our defensive JSON checks to enable nullable/optional fields by adding an optional param to `util/json.py`

## Test Coverage
Made a POST request to the endpoint locally both with/without eatery_id (int) provided to make sure it works along with required fields type (str) and content (str)
Screen Shot 2022-04-18 at 8 52 39 PM

To test:
run `bash reset_db.sh` to apply migrations, then make a POST request to /report with required params type and content and optional param eatery_id

## Screenshots (delete if not applicable)

<details>

  <summary>Pic of feature that this PR is needed for</summary>

Screen Shot 2022-04-18 at 9 04 51 PM

</details>

### CODE REVIEW

File: src/api/util/json.py [hunk 1]
```diff
@@ -22,19 +25,13 @@ def verify_json_fields(json, field_type_map: Mapping[str, FieldType]):
...
     return True
 
+
 def success_json(data):
-    return {
-        "success": True,
-        "data": data,
-        "error": None
-    }
+    return {"success": True, "data": data, "error": None}
```
Comment (alanna-zhou) [inline src/api/util/json.py:33]: this is a bit... unconventional?

i know you did not write this code, but response fields should either be
```
{
  "success": True, "data": data
}
```

or

```
{
  "success": True, "error": error
}
```

i've never seen this kind of response lol

File: src/api/util/json.py [hunk 2]
```diff
@@ -22,19 +25,13 @@ def verify_json_fields(json, field_type_map: Mapping[str, FieldType]):
         elif field_type_map[field] is FieldType.EATERYID:
             if not isinstance(json[field], int) or EateryID(json[field]) == None:
                 return False
-            
+
     return True
 
+
 def success_json(data):
```
Comment (alanna-zhou) [inline src/api/util/json.py:32]: this hsould also be `success_response`

Comment (alanna-zhou) [reply src/api/util/json.py:32]: but ik you didn't write the code.. up to you on if you want to refactor later

Comment (alanna-zhou) [reply src/api/util/json.py:32]: but yeah `success_response` is appdev standard

**Review by alanna-zhou — APPROVED** [review]
nice! it makes sense

File: src/api/util/json.py (hunk 2 above)
Comment (Archit404Error) [reply src/api/util/json.py:32]: Gotchu, yeah android wants the feature ASAP so I'll merge this for now and remember to refactor this soon!

**Pull request merged**